api/v1alpha1/validate_routes.go

```diff
@@ -0,0 +1,35 @@
+/*
+ Copyright 2024, NVIDIA CORPORATION & AFFILIATES
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"net"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// validateRoutes validate routes:
+// - dst is a valid CIDR
+func validateRoutes(routes []Route, fldPath *field.Path) field.ErrorList {
```

Comment:
I think we should also validate that routes are from the same address family (ipv4, ipv6) as the pool

Reply:
Done